src/easylink/step.py

```diff
@@ -29,212 +40,44 @@
-                if mapping.parent_slot == edge.output_slot
-            ]
-            for mapping in mappings:
-                new_edge = mapping.remap_edge(edge)
-                new_step = self._step.step_graph.nodes[mapping.child_node]["step"]
-                imp_edges = new_step.get_implementation_edges(new_edge)
-                implementation_edges.extend(imp_edges)
-
-        elif edge.target_node == self._step.name:
-            mappings = [
-                mapping
-                for mapping in self._step.slot_mappings["input"]
-                if mapping.parent_slot == edge.input_slot
-            ]
-            for mapping in mappings:
-                new_edge = mapping.remap_edge(edge)
-                new_step = self._step.step_graph.nodes[mapping.child_node]["step"]
-                imp_edges = new_step.get_implementation_edges(new_edge)
-                implementation_edges.extend(imp_edges)
-        else:
-            raise ValueError(f" {self._step.name} not in edge {edge}")
-        if not implementation_edges:
-            raise ValueError(f"No edges found for {self._step.name} in edge {edge}")
-        return implementation_edges
-
-    def configure_subgraph_steps(self) -> None:
-        for node in self._step.step_graph.nodes:
-            step = self._step.step_graph.nodes[node]["step"]
-            step.set_configuration_state(
-                self.pipeline_config, self.combined_implementations, self.input_data_config
-            )
-
-
 class Step:
     """The highest-level pipeline building block abstraction.
 
-    Steps contain information about the purpose of the interoperable elements of
-    the sequence called a *Pipeline* and how those elements relate to one another.
-    In turn, steps are implemented by Implementations, such that each step may have
-    several implementations but each implementation must have exactly one step.
-    In a sense, steps contain metadata about the implementations to which they relate.
+    ``Steps`` contain information about the purpose of the interoperable elements of
+    the sequence called a "pipeline" and how those elements relate to one another.
+    In turn, ``Steps`` are implemented by :class:`Implementations<easylink.implementation.Implementation>`,
+    such that each ``Step`` may have several ``Implementations`` to choose from
+    but each ``Implementation`` must implemement exactly one ``Step`. As such, the
+    pipeline for a given EasyLink run consists of ``Implementations`` that collectively
+    span the ``Steps`` in the :class:`~easylink.pipeline_schema.PipelineSchema`.
 
-    Attributes
+    Parameters
     ----------
-    _configuration_state
-        The :class:`~easylink.step.ConfigurationState`.
+    step_name
+        The name of the high-level pipeline step.
+    name
+        The name of the step *node*. This is used to disambiguate names as necessary,
+        e.g. if "step 1" is looped multiple times.
```

Comment:
```suggestion
    step_name
        The name of the pipeline step in the ``PipelineSchema``.
    name
        The name of this step *node*. This can be different from the step_name
        due to the need for disambiguation during the process of unrolling loops, etc.
        For example, if "step 1" were looped multiple times, each node would have
        a step_name of "step 1" but unique names.
```

Reply:
The last "name" in this suggestion should probably be code-formatted so it is clearer I'm talking about the actual attribute. And maybe by extension `step_name` should be as well.